Hw 2 solution

Iskamele opened the pull request.

Iskamele committed:
Added a constant with the number of balls.
Added a loop that displays the result of the getRandomBall method to the console.

Iskamele committed:
As part of the concept, an enum was created containing colors for the balls.

Iskamele committed:
Added a constant with the maximum allowable value on the ball.
Added the getRandomBall method, inside which the Ball object is created.
The getRandomColor method sets the color value for the Ball and sets a random number that does not exceed the BALL_NUMBER constant. The method returns an initialized object.

Iskamele committed:
Added getRandomColor method, inside which color is randomly selected, which is provided from the total number of colors in the Colors enum.

Iskamele committed:
Added private fields for color and ball number, and setters for them.
The toString method has been overridden so that it outputs a string response to the console containing the color and number on the ball.

Rommelua reviewed (changes requested):
![image](https://user-images.githubusercontent.com/20326858/198889516-3b618869-48e9-4612-902c-0b54fb186282.png)
Java Cl checks failed. Before making a Pull Request (or a new commit to an already existing PR), you need to write the mvn clean package command in the terminal and make sure that all checks pass. After that, the mentor will be able to review the homework

Iskamele committed:
Added a newlines at the end of files

Iskamele commented:
I added the missing spaces at the end of the classes, pushed it, but I don’t see relevant comments in the pull request, as well as reactions to changes in general. I see first commit
![image](https://user-images.githubusercontent.com/74711003/198895977-62a0429f-9a0f-4fdd-9e97-b39e5c06e8f3.png)
![image](https://user-images.githubusercontent.com/74711003/198895979-6ab9a3bd-b5de-4797-87ab-bfdec6cdb93b.png)
![image](https://user-images.githubusercontent.com/74711003/198896026-1ec54385-9ed7-4238-bdc3-ee613aff178e.png)

Iskamele committed:
Removed comment

Iskamele reviewed (commented):
Do I need to change the return type of the getRandomColor method?
I just now noticed that I did not use String as a result, but replaced it with Colors.values()[index]

![image](https://user-images.githubusercontent.com/74711003/198899582-d4a0b6dd-0299-4d29-9c59-7a97e9b17e90.png)

Iskamele committed:
Changed the return data type in the getRandomColor() method from Colors to String, as it was in the draft.
Random enum color is converted to a string using toString().

Iskamele committed:
The private field Colors color has been changed to String.
Accordingly, the setter was changed for a different data type.

Iskamele commented:
Changed the data type in the method in the Ball class. Waiting for an answer

aaananas commented on src/main/java/core/basesyntax/Lottery.java:
```diff
@@ -0,0 +1,14 @@
+
+    public Ball getRandomBall() {
+        Ball ball = new Ball();
+        ball.setColor(new ColorSupplier().getRandomColor());
+        ball.setNumber(new Random().nextInt(BALL_NUMBER));
```
lets make Random and ColorSupplier class-level variables

Iskamele replied:
Got it, done

aaananas commented on src/main/java/core/basesyntax/ColorSupplier.java:
```diff
@@ -1,7 +1,10 @@
 public class ColorSupplier {
     public String getRandomColor() {
-        return null;
+        int index = new Random().nextInt(Colors.values().length);
```
lets make Random class-level variable

Iskamele replied:
I did but did not understand why this is necessary and what is the difference. Wrote this question for Q&A on Saturday

aaananas reviewed (changes requested):
great job

Iskamele committed:
Made the color and number fields final, as recommended by the compiler.
Replaced setters with a constructor for these fields.

Iskamele committed:
Made Random class-level variable.
Replaced toString() with name().

Iskamele committed:
Made Random class-level variable.
Changed the getRandomBall() method so that it uses the received random values and creates an object using the constructor from the Ball class.

zachar-chorny reviewed (approved):
Already approved
